packages/server/src/api/controllers/datasource.ts

```diff
@@ -441,3 +442,18 @@ export async function query(ctx: UserCtx) {
     ctx.throw(400, err)
   }
 }
+
+export async function getExternalSchema(ctx: Ctx) {
```

Comment:
This should probably be `UserCtx` rather than `Ctx` as there will be a user available